src/move/econia/sources/user.move

```diff
@@ -409,6 +409,78 @@ module econia::user {
         size: u64
     }
 
+    /// Table keys are market account IDs.
+    struct MarketEventHandles has key {
+        cancel_posted_order_events:
+            Tablist<u128, EventHandle<CancelPostedOrderEvent>>,
+        change_order_size_events:
+            Tablist<u128, EventHandle<ChangeOrderSizeEvent>>,
+        fill_events: Tablist<u128, EventHandle<FillEvent>>,
+        place_limit_order_events:
+            Tablist<u128, EventHandle<PlaceLimitOrderEvent>>,
+        place_market_order_events:
+            Tablist<u128, EventHandle<PlaceMarketOrderEvent>>
+    }
+
+    struct CancelPostedOrderEvent has copy, drop, store {
+        market_id: u64,
+        order_id: u128,
+        user: address,
+        custodian_id: u64,
+        side: bool,
+        is_eviction: bool,
+        is_self_match_cancel: bool
```

Comment:
It's more common to simply have a `reason` field. See for example
<https://docs.cloud.coinbase.com/exchange/docs/websocket-channels#done>

This would be more in line with industry standards and offers greater flexibility and extensibility.

Reply:
See https://github.com/econia-labs/econia/pull/321#issuecomment-1632592184